connector-luna/src/main/java/com/datastax/oss/pulsar/source/converters/NativeAvroConverter.java

```diff
@@ -180,64 +199,92 @@ Schema buildUDTSchema(KeyspaceMetadata ksm, String typeName, boolean optional) {
+                        genericRecordBuilder.put(fieldName, row.getInstant(cm.getName()).toEpochMilli());
                         break;
                     case ProtocolConstants.DataType.DATE: // Avro date is epoch days
-                        genericRecordBuilder.put(cm.getName().toString(), (int) row.getLocalDate(cm.getName()).toEpochDay());
+                        genericRecordBuilder.put(fieldName, (int) row.getLocalDate(cm.getName()).toEpochDay());
                         break;
                     case ProtocolConstants.DataType.TIME: // Avro time is epoch milliseconds
-                        genericRecordBuilder.put(cm.getName().toString(), (row.getLocalTime(cm.getName()).toNanoOfDay() / 1000));
+                        genericRecordBuilder.put(fieldName, (row.getLocalTime(cm.getName()).toNanoOfDay() / 1000));
                         break;
                     case ProtocolConstants.DataType.BLOB:
-                        genericRecordBuilder.put(cm.getName().toString(), row.getByteBuffer(cm.getName()));
+                        genericRecordBuilder.put(fieldName, row.getByteBuffer(cm.getName()));
                         break;
                     case ProtocolConstants.DataType.UDT:
-                        genericRecordBuilder.put(cm.getName().toString(), buildUDTValue(row.getUdtValue(cm.getName())));
+                        genericRecordBuilder.put(fieldName, buildUDTValue(row.getUdtValue(cm.getName())));
                         break;
                     case ProtocolConstants.DataType.DURATION:
-                        genericRecordBuilder.put(cm.getName().toString(), row.getCqlDuration(cm.getName()));
+                        genericRecordBuilder.put(fieldName, row.getCqlDuration(cm.getName()));
                         break;
                     case ProtocolConstants.DataType.DECIMAL:
-                        genericRecordBuilder.put(cm.getName().toString(), row.getBigDecimal(cm.getName()));
+                        genericRecordBuilder.put(fieldName, row.getBigDecimal(cm.getName()));
                         break;
                     case ProtocolConstants.DataType.VARINT:
-                        genericRecordBuilder.put(cm.getName().toString(), row.getBigInteger(cm.getName()));
-                        break;
+                        genericRecordBuilder.put(fieldName, row.getBigInteger(cm.getName()));
+                        break;
+                    case ProtocolConstants.DataType.LIST: {
+                        ListType listType = (ListType) cm.getType();
+                        Schema listSchema = collectionSchemas.get(fieldName);
+                        List listValue = row.getList(fieldName, CodecRegistry.DEFAULT.codecFor(listType.getElementType()).getJavaType().getRawType());
+                        log.info("field={} listSchema={} listValue={}", fieldName, listSchema, listValue);
```

Comment:
this log should be removed perhaps

Reply:
yes, for example, cloudbeds has list<map<text, int>>, so I still need to add integration tests for such nested tricky use-case, then I will remove comments...